src/Mvc/Mvc.Razor.RuntimeCompilation/src/AssemblyPartResolvers.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation
+{
+    /// <summary>
+    /// This class is a workaround to enable compilation of plugin references which DependencyContext is unaware of
+    /// </summary>
+    internal class DependencyContextResolver : IAssemblyPartResolver
+    {
+        public IEnumerable<string> GetReferencePaths(AssemblyPart assemblyPart)
+        {
+            IEnumerable<string> references;
+            try
+            {
+                references = assemblyPart.GetReferencePaths();
+            }
+            catch (InvalidOperationException)
+            {
+                // DependencyContext was unable to resolve the dependencies because it is unaware of additional references.
+                yield break;
+            }
+
+            foreach (var reference in references)
+            {
+                yield return reference;
```

Comment:
I feel like I'm missing something here: How is this different than just saying `return references;` and changing the `yield break;` above to a `return null;` or `return Array.Empty<string>();` or similar?